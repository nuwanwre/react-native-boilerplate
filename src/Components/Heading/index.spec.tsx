// #region Global Imports
import * as React from "react";
import { renderWithProviders } from "@Test/Helpers/render";
import { shallow } from "enzyme";
// #endregion Global Imports

// #region Local Imports
import { Heading } from "@Components";
// #endregion Local Imports

describe("Components", () => {
    describe("Heading", () => {
        // const wrap = renderer.create(<Heading text="World" />).toJSON();

        it("should match snapshot", () => {
            const wrapper = renderWithProviders(<Heading text="World" />);

            expect(wrapper).toMatchSnapshot();
        });

        const wrap = shallow(<Heading text="World" />);

        it("should render FrontText and  BackText without throwing an error", () => {
            expect(wrap.children().length).toEqual(2);
        });

        it("FrontText should equal World", () => {
            expect(
                wrap
                    .children()
                    .first()
                    .text()
            ).toEqual("World");
        });
    });
});
