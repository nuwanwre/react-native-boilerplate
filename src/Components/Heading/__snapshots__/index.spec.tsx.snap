// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Heading should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "relative",
        "textAlign": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 35,
          "fontWeight": "bold",
          "position": "absolute",
        },
      ]
    }
  >
    World
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 80,
          "fontWeight": "bold",
          "letterSpacing": 1,
          "lineHeight": 86,
          "zIndex": -1,
        },
      ]
    }
  >
    World
  </Text>
</View>
`;
